src/components/auth/qrSignIn.tsx

```diff
@@ -0,0 +1,76 @@
+import { QueryClientContextProvider } from '$constants/queryClient'
+import { useQuery } from '@tanstack/react-query'
+import { useEffect, useRef } from 'react'
+
+export interface QrSignIn {
+  state: string
+}
+export function QrSignIn(props: QrSignIn) {
+  return (
+    <QueryClientContextProvider>
+      <QrSignInView {...props} />
+    </QueryClientContextProvider>
+  )
+}
+function QrSignInView(props: QrSignIn) {
+  const { data } = useQuery({
+    queryKey: ['qrSignIn'],
+    refetchOnWindowFocus: false,
+    queryFn: async () => {
+      const deviceId = crypto.randomUUID()
+      const url = `${location.origin}/auth/mobile?deviceId=${deviceId}`
```

Comment:
An attacker who sees the QR code may be able to hijack the signature as the same `deviceId` can be used to query the signature. Hash this.